unitylibs/core/workflow/workflow-firefly/widget.js

```diff
@@ -508,9 +579,22 @@ export default class UnityWidget {
     if (Array.isArray(data)) {
       data.forEach((item) => {
         const itemEnv = item.env || 'prod';
-        if (item.verb && item.prompt && item.assetid && itemEnv === unityConfig.env) {
+        if (item.verb && item.prompt && itemEnv === unityConfig.env) {
           if (!promptMap[item.verb]) promptMap[item.verb] = [];
-          promptMap[item.verb].push({ prompt: item.prompt, assetid: item.assetid });
+          let variations = [];
+          if (item.verb === 'sound') {
+            const ph = this.workflowCfg?.placeholder || {};
+            const labels = Object.keys(ph)
+              .filter((k) => k.startsWith('placeholder-variation-label-'))
+              .map((k) => ({ idx: Number.parseInt(k.split('-').pop(), 10), val: (ph[k] || '').trim() }))
+              .filter(({ idx, val }) => Number.isFinite(idx) && val)
+              .sort((a, b) => a.idx - b.idx)
+              .map(({ val }) => val);
+            const urls = (typeof item.variationUrls === 'string' ? item.variationUrls : '')
+              .split('||').map((s) => s.trim()).filter((s) => s);
+            variations = labels.map((lbl, idx) => ({ label: lbl, url: urls[idx] || '' }));
+          }
+          promptMap[item.verb].push({ prompt: item.prompt, assetid: item.assetid, variations });
```

Comment:
Nit: Same thing here? I think it creates a good pattern if new verbs require different logic.